tools/cell_census_builder/scripts/release/census_summary.py

```diff
@@ -28,3 +30,64 @@
     stats_df = pd.DataFrame(stats, columns=["organism", "attribute", "unique count"])
     display_stats_df = pd.pivot(stats_df, index=["organism"], columns=["attribute"], values=["unique count"])
     print(display_stats_df)
+
+    # Calculate the diff with respect to the previous version (if specified)
+
+    if previous_census_version is not None:
+        previous_census = cell_census.open_soma(census_version=previous_census_version)
+
+        prev_datasets = previous_census[CENSUS_INFO_NAME]["datasets"].read().concat().to_pandas()
+        curr_datasets = census[CENSUS_INFO_NAME]["datasets"].read().concat().to_pandas()
+
+        # Datasets removed, added
+        curr_datasets_ids = set(curr_datasets["dataset_id"])
+        prev_dataset_ids = set(prev_datasets["dataset_id"])
+
+        added_datasets = curr_datasets_ids - prev_dataset_ids
+        removed_datasets = prev_dataset_ids - curr_datasets_ids
+        if added_datasets:
+            print("Datasets that were added")
+            for d in added_datasets:
+                print(d)
+
+        if removed_datasets:
+            print("Datasets that were removed")
+            for d in removed_datasets:
+                print(d)
+
+        # Datasets in both versions but that have differing cell counts
+        joined = prev_datasets.join(
+            curr_datasets.set_index("dataset_id"), on="dataset_id", lsuffix="_prev", rsuffix="_curr"
+        )
+        datasets_with_different_cell_counts = joined.loc[
+            joined["dataset_total_cell_count_prev"] != joined["dataset_total_cell_count_curr"]
+        ][["dataset_id", "dataset_total_cell_count_prev", "dataset_total_cell_count_curr"]]
+
+        if not datasets_with_different_cell_counts.empty:
+            print("Datasets that have a different cell count")
+            print(datasets_with_different_cell_counts)
+
+        # Total cell count deltas by experiment (mouse, human)
+        for organism in ["homo_sapiens", "mus_musculus"]:
+            curr_count = census[CENSUS_DATA_NAME][organism]["obs"].count
+            prev_count = previous_census[CENSUS_DATA_NAME][organism]["obs"].count
+            print(
+                f"Previous {organism} cell count: {prev_count}, current {organism} cell count: {curr_count}, delta {curr_count - prev_count}"
+            )
+
+        # Deltas between summary_cell_counts dataframes
```

Comment:
I have the code for this but the dataframe here can be quite big. The current runs shows 65 different entries for total cell counts. Do we want to display it regardless? An alternative is to only show the deltas that are bigger than a certain threshold.

Reply:
that is a Product question - I don't know exactly how they intend to use this.  @pablo-gar - please specify your desired report format.

My own opinion:  the point of this is to summarize _all_ major changes, so we want to list all of the differences.

Reply:
I added them to the code.